server/api/utils/scheduler.py

```diff
@@ -944,6 +947,79 @@ def _enrich_schedule(
             labels.setdefault(mlrun_constants.MLRunInternalLabels.kind, fn_kind)
         return labels
 
+    @staticmethod
+    def _merge_schedule_and_schedule_object_labels(
+        labels: Optional[dict],
+        scheduled_object: Union[Optional[dict], Callable],
+    ):
+        """
+        Merges the labels of the scheduled object, giving precedence to the scheduled object labels
+        :param labels: the labels of a schedule
+        :param scheduled_object: a scheduled object
+
+        :return: merged labels
+        """
+        # Ensure scheduled_object is a dictionary-like object
+        if not hasattr(scheduled_object, "get"):
```

Comment:
More accurate/readable
```suggestion
        if not isinstance(scheduled_object, dict):
```